packages/ckeditor5-html-support/docs/features/general-html-support.md

```diff
@@ -0,0 +1,212 @@
+	The General HTML support feature is **experimental and not yet production-ready**.
+
+	Follow the ["Stabilize and release a production-ready General HTML support feature"](https://github.com/ckeditor/ckeditor5/issues/9856) issue for more updates and related issues.
+</info-box>
+
+## Demo
+
+The general HTML support feature is configured via the `config.htmlSupport` property. In it, you need to list the HTML features that should be handled by GHS.
+
+The demo below uses the following configuration:
+
+```js
+import SourceEditing from '@ckeditor/ckeditor5-source-editing/src/sourceediting';
+import GeneralHtmlSupport from '@ckeditor/ckeditor5-html-support/src/generalhtmlsupport';
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ SourceEditing, GeneralHtmlSupport, ... ],
+		toolbar: [ 'sourceEditing', ... ],
+		htmlSupport: {
+			allow: {
+				// Enables <div>, <details>, and <summary> elements with all kind of attributes.
+				{
+					name: /^(div|details|summary)$/,
+					styles: true,
+					classes: true,
+					attributes: true
+				},
+
+				// Extends the existing Paragraph and Heading features
+				// with classes and data-* attributes.
+				{
+					name: /^(p|h[2-4])$/,
+					classes: true,
+					attributes: /^data-/
+				},
+
+				// Enables <span>s with any inline styles.
+				{
+					name: 'span',
+					styles: true
+				},
+
+				// Enables <abbr>s with the title attribute.
+				{
+					name: 'abbr',
+					attributes: [ 'title' ]
+				}
+			}
+		}
+	} )
+	.then( ... )
+	.catch( ... );
+```
+
+{@snippet features/general-html-support}
+
+## Level of support
+
+The difference between specific CKEditor 5 features such as {@link features/basic-styles basic styles} or {@link features/headings headings} and the HTML features enabled by GHS is that a plugin that supports a specific HTML feature provides a complete user experience for that feature, whereas GHS ensures only that such a content is accepted by the editor.
+
+For instance, the dedicated {@link features/basic-styles#available-text-styles bold} feature offers a toolbar button used to make the selected text bold. Together with the {@link features/autoformat autoformatting feature}, it also allows for applying bold style to content by typing a Markdown shortcode (`**foo**`) in the editor. The {@link features/headings headings} feature offers a dropdown from which the user can choose a heading level and ensures that pressing <kbd>Enter</kbd> at the end of a heading creates a new paragraph (and not another heading).
+
+The general HTML support does not offer any UI for the enabled features and takes only basic semantics of a given feature into account. If you enable support for `<div>`s via GHS, the user will not be able to create `<div>`s from the editor UI. The GHS will know that a `<div>` is a container element, so it can wrap other blocks (like paragraphs) but cannot be used inline (next to e.g. a `<strong>` element). There is no more, however.
```

Comment:
```suggestion
The General HTML Support does not offer any UI for the enabled features and takes only the basic semantics of a given feature into account. If you enable support for `<div>`s via GHS, the user will not be able to create `<div>`s from the editor UI. The GHS will know that a `<div>` is a container element, so it can wrap other blocks (like paragraphs) but cannot be used inline (next to e.g. a `<strong>` element). There is no more, however.
```